### ISSUE TITLE
[ERROR] TypeError: request.defaults is not a function

### ISSUE THREAD

**Issue by roblav96** [voltrue2/in-app-purchase#42]
Just updated to the latest version, and I'm getting this error when I start my server:

```
R-0 =============================== ERROR ================================
R-0 > ERR > 12:55:57:164[/usr/local/lib/node_modules/pm2/node_modules/pmx/lib/notify.js:47]
R-0         TypeError: request.defaults is not a function
R-0     at Object.module.exports.readConfig (/Users/server/Projects/app/server/node_modules/in-app-purchase/lib/apple.js:41:21)
R-0     at Object.module.exports.config (/Users/server/Projects/app/server/node_modules/in-app-purchase/index.js:15:8)
R-0     at Object.<anonymous> (/Users/server/Projects/app/server/server/controllers/iap-ctrl.js:15:5)
R-0     at Module._compile (module.js:413:34)
R-0     at Object.Module._extensions..js (module.js:422:10)
R-0     at Module.load (module.js:357:32)
R-0     at Function.Module._load (module.js:314:12)
R-0     at Function._load (/usr/local/lib/node_modules/pm2/node_modules/pmx/lib/transaction.js:62:21)
R-0     at Module.require (module.js:367:17)
R-0     at require (internal/module.js:16:19)
R-0     at Object.<anonymous> (/Users/server/Projects/app/server/server/routes/iap/verify.js:13:15)
R-0     at Module._compile (module.js:413:34)
R-0     at Object.Module._extensions..js (module.js:422:10)
R-0     at Module.load (module.js:357:32)
R-0     at Function.Module._load (module.js:314:12)
R-0     at Function._load (/usr/local/lib/node_modules/pm2/node_modules/pmx/lib/transaction.js:62:21)
R-0     at Module.require (module.js:367:17)
R-0     at require (internal/module.js:16:19)
R-0     at Object.<anonymous> (/Users/server/Projects/app/server/server/server.js:287:33)
R-0     at Module._compile (module.js:413:34)
R-0     at Object.Module._extensions..js (module.js:422:10)
R-0     at Module.load (module.js:357:32)
PM2 App [R] with id [0] and pid [700], exited with code [255] via signal [null]
```

My controller at
```
iap-ctrl.js:15:5
```
```javascript
iap.config( {
	requestDefaults: {
		timeout: 10000
	},
	applePassword: "1337", // this comes from iTunes Connect
	googlePublicKeyStrSandbox: "1337", // this is the path to the directory containing iap-sanbox/iap-live files
	googlePublicKeyStrLive: "1337" // this is the path to the directory containing iap-sanbox/iap-live files
} )
iap.setup()
```

**Comment by roblav96** [comment]
Reverting back to
```
in-app-purchase@0.7.4
```
fixes the problem.

**Comment by voltrue2** [comment]
Hello

Thank you for reporting about this problem.

I have fixed the issue and published the fixed version.

It's v1.0.1

**Comment by roblav96** [comment]
awesome! thanks man!

**Comment by voltrue2** [comment]
Hello Rob

Let me have a look into the issue.

**Comment by voltrue2** [comment]
@roblav96 So the issue is that `iap.setup()` actually expects a callback.
The previous version tolerated missing callback, but update I did actually made it so that it **MUST** have a callback.
I realized that it is better for the package to handle the way it used to meaning it should not throw an error when it sees no callback.

I have updated that part of the code and published new version `1.0.2`.

I apologize for the sudden breakage of the module, but thank you for taking the time to report it :-)
